shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ShenyuClientRegisterRepositoryFactory.java

```diff
@@ -29,9 +30,32 @@
  * The type shenyu client register repository factory.
  */
 public final class ShenyuClientRegisterRepositoryFactory {
-    
+
     private static final Map<String, ShenyuClientRegisterRepository> REPOSITORY_MAP = new ConcurrentHashMap<>();
-    
+
+    /**
+     * Get the REPOSITORY_MAP.
+     *
+     * @return the REPOSITORY_MAP
+     */
+    //getter
+    public static Map<String, ShenyuClientRegisterRepository> getRepositoryMap() {
+        return Collections.unmodifiableMap(REPOSITORY_MAP);
+    }
+
+    /**
+     * Sets the repository map with the provided map of ShenyuClientRegisterRepository instances.
+     * Clears the existing repository map and adds all entries from the provided map.
+     *
+     * @param repositoryMap The map containing ShenyuClientRegisterRepository instances,
+     *                      where each entry's key is the repository type identifier.
+     */
+    // Setter for testing purposes
```

Comment:
why set?

Reply:
I am trying to use call `getRepositoryMap().clear();` but I am facing this error `Immutable object cannot be modified` even though I am using the getter so to fix this I read somewhere it can be fixed using setter along with  getter to create a new map `ConcurrentHashMap<>()` for testing purposes but that also didin't end up working. It's also the same for the .put() in `getRepositoryMap()).put("someType", repositoryMock)` @moremind